cpp/ql/src/semmle/code/cpp/models/implementations/Allocation.qll

```diff
@@ -14,19 +14,24 @@ private class MallocAllocationFunction extends AllocationFunction {
   int sizeArg;
 
   MallocAllocationFunction() {
+    // --- C library allocation
     hasGlobalOrStdName("malloc") and // malloc(size)
     sizeArg = 0
     or
     hasGlobalName([
+        // --- Windows Memory Management for Windows Drivers
         "MmAllocateContiguousMemory", // MmAllocateContiguousMemory(size, maxaddress)
         "MmAllocateContiguousNodeMemory", // MmAllocateContiguousNodeMemory(size, minaddress, maxaddress, bound, flag, prefer)
         "MmAllocateContiguousMemorySpecifyCache", // MmAllocateContiguousMemorySpecifyCache(size, minaddress, maxaddress, bound, type)
         "MmAllocateContiguousMemorySpecifyCacheNode", // MmAllocateContiguousMemorySpecifyCacheNode(size, minaddress, maxaddress, bound, type, prefer)
         "MmAllocateNonCachedMemory", // MmAllocateNonCachedMemory(size)
         "MmAllocateMappingAddress", // MmAllocateMappingAddress(size, tag)
+        // --- Windows COM allocation
         "CoTaskMemAlloc", // CoTaskMemAlloc(size)
+        // --- Linux kernel memory allocator
         "kmem_alloc", // kmem_alloc(size, flags)
         "kmem_zalloc", // kmem_zalloc(size, flags)
```

Comment:
I don't think these are from Linux. [The Google results for them](https://www.google.com/search?q=kmem_alloc) refer to Solaris and BSD. Where they mention Linux, it's in the context of ZFS or XFS: file systems that were ported to Linux from other kernels and therefore use wrapper functions.

My best suggestion is to write `Solaris/BSD kernel memory allocator` here.

Reply:
Thanks.  Done.